workitem/link/link_repository.go

```diff
@@ -88,16 +96,28 @@ func (r *GormWorkItemLinkRepository) ValidateCorrectSourceAndTargetType(ctx cont
 		return errs.WithStack(err)
 	}
 	// Check type paths
```

Comment:
> From my understanding, a link type combination is valid if the source and target link types match the ones described in the combination or if they are subtypes of the ones described in the combination. Correct ?

Correct.

> BTW, if this is the case, did you add tests to verify that this validation method behaves properly in all cases, including with valid and invalid subtypes ?

Inside of `ValidateCorrectSourceAndTargetType()` I use `WIT.IsTypeOrSubtypeOf` which does the main work. That function is tested [here](https://github.com/almighty/almighty-core/blob/52a313e34c38fc764e8b8057690964a91ee66787/workitem/workitemtype_blackbox_test.go#L217).